src/lib/geolocation.ts

```diff
@@ -0,0 +1,395 @@
+            'SD',
+            'SG',
+            'SL',
+            'SN',
+            'SO',
+            'SR',
+            'SS',
+            'ST',
+            'SV',
+            'SX',
+            'SY',
+            'SZ',
+            'TC',
+            'TD',
+            'TG',
+            'TH',
+            'TJ',
+            'TK',
+            'TL',
+            'TM',
+            'TN',
+            'TO',
+            'TT',
+            'TW',
+            'TZ',
+            'UA',
+            'UG',
+            'UM',
+            'UY',
+            'UZ',
+            'VC',
+            'VE',
+            'VG',
+            'VI',
+            'VN',
+            'VU',
+            'WS',
+            'YE',
+            'ZA',
+            'ZM',
+            'ZW',
+        ],
+    },
+    NZDCountries: {
+        name: 'New Zealand',
+        currency: 'NZD',
+        countries: ['NZ', 'CK'],
+    },
+    Canada: {
+        name: 'Canada',
+        currency: 'CAD',
+        countries: ['CA'],
+    },
+};
+
+type CountryNameMap = Record<string, string>;
+
+const countryNames: CountryNameMap = {
+    GB: 'the UK',
+    US: 'the US',
+    AU: 'Australia',
+    CA: 'Canada',
+    DE: 'Germany',
+    NZ: 'New Zealand',
+    FR: 'France',
+    NL: 'the Netherlands',
+    IE: 'Ireland',
+    SE: 'Sweden',
+    CH: 'Switzerland',
+    NO: 'Norway',
+    BE: 'Belgium',
+    IT: 'Italy',
+    IN: 'India',
+    ES: 'Spain',
+    DK: 'Denmark',
+    SG: 'Singapore',
+    AT: 'Austria',
+    FI: 'Finland',
+    HK: 'Hong Kong',
+    LU: 'Luxembourg',
+    PT: 'Portugal',
+    AE: 'the UAE',
+    MX: 'Mexico',
+    BR: 'Brazil',
+    ZA: 'South Africa',
+    TW: 'Taiwan',
+    IL: 'Israel',
+    JP: 'Japan',
+    CZ: 'the Czech Republic',
+    GR: 'Greece',
+    IS: 'Iceland',
+    TH: 'Thailand',
+    MY: 'Malaysia',
+    RO: 'Romania',
+    PL: 'Poland',
+    HU: 'Hungary',
+    TR: 'Turkey',
+    KR: 'Korea',
+    SI: 'Slovenia',
+    CL: 'Chile',
+    CO: 'Colombia',
+    QA: 'Qatar',
+    HR: 'Croatia',
+    SK: 'Slovakia',
+    ID: 'Indonesia',
+    VN: 'Vietnam',
+    CN: 'China',
+    MT: 'Malta',
+    AR: 'Argentina',
+    KE: 'Kenya',
+    PR: 'Puerto Rico',
+    RU: 'Russia',
+    EE: 'Estonia',
+    CR: 'Costa Rica',
+    PA: 'Panama',
+};
+
+const extendedCurrencySymbol = {
+    GBPCountries: '£',
+    UnitedStates: '$',
+    AUDCountries: '$',
+    Canada: 'CA$',
+    EURCountries: '€',
+    NZDCountries: 'NZ$',
+    International: '$',
+};
+
+const countryCodeToCountryGroupId = (countryCode: string): CountryGroupId => {
+    const availableCountryGroupIds = Object.keys(countryGroups) as CountryGroupId[];
+
+    const foundCountryGroupId = availableCountryGroupIds.find(countryGroupId =>
+        countryGroups[countryGroupId].countries.includes(countryCode),
+    );
+
+    return foundCountryGroupId || 'International';
+};
+
+const defaultCurrencySymbol = '£';
+
+// There's an interesting issue here where the default currency symbol (£) is
```

Comment:
Maybe be even more explicit and state 'we think this is a bug but porting as is for now during migration' (or some equivalent)?

Reply:
I've updated the comment based on your suggestion. But I do think this logic might be intentional, at least when I read it it kind of made sense to me.
When we don't know where the user is (could be in the UK or not) we serve the Guardian's local currency. When we know where they are but can't figure out the country name (definitely not the UK) we use USD as an international currency.

Reply:
Ah yeah good shout - that could definitely be the reason!